src/unitxt/processors.py

```diff
@@ -277,3 +279,57 @@ def process_value(self, text: Any) -> Any:
 
         except:
             return 0
+
+
+class InferDictsToBinaryLogprobs(FieldOperator):
+    binary_class_names: list[str]
+
+    def process_value(self, obj: Any) -> Any:
+        try:
+            n_first_tokens = np.min([3, len(obj)])
+        except:
+            return []
+
+        for i in range(n_first_tokens):
+            try:
+                pos_probs, neg_probs = self.get_pos_neg_probs(pred_dict=obj[i])
+                if pos_probs or neg_probs:
+                    sum_probs = sum(pos_probs) + sum(neg_probs)
+                    if sum_probs > 0.0001:
+                        return sum(pos_probs) / sum_probs
+            except:
+                pass
+        return []
+
+    def get_pos_neg_probs(self, pred_dict):
+        token_logprobs = pred_dict["top_tokens"]
+        pos_probs = [
+            np.exp(d["logprob"])
+            for d in token_logprobs
+            if self.binary_class_names[1].lower() in d["text"].lower()
```

Comment:
What happens is the token text  is "noone" - it will include "no" in the answer.

Reply:
Yes, this can happen, although not super likely given a prompt that asks for yes or no.
But also, there are several different alternatives of how to say yes and no depending on the model as well as the tokenizer, and often there are special tokens somewhere in there because it is the first word. Specifically tokens that start with space may appear in various ways for different tokenizers and inference APIs. Probably a more complicated regex would be needed to cover all of them.

Reply:
I also saw tokens with spaces or period.  That's why a minimal regex is needed.

Reply:
so I guess something more like `^(\W|\\n)?[yY]es(\W|\\n)?$`? not sure

Reply:
done